Documentation/update-codeowners.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+source_dir="$(cd "${script_dir}/.." && pwd)"
+
+echo "<!-- This file was autogenerated via Documentation/update-codeowners.sh, do not edit manually-->" \
+    > "${script_dir}/codeowners.rst"
+echo >> "${script_dir}/codeowners.rst"
+sed '/END_CODEOWNERS_DOCS/q' "${source_dir}/CODEOWNERS" \
+    | head -n-2 \
+    | cut -c 3- \
+    | sed 's;@cilium/\([-_a-zA-Z0-9]*\)\([:]*\);`@cilium/\1 <https://github.com/orgs/cilium/teams/\1>`_\2;g' \
```

Comment:
Let's use anonymous links to avoid creating multiple targets with the same name. CI complains about it by the way.

The square brackets around the colon in the sed expression also seem unnecessary.

(As a side note, `;` is not the best choice for a delimiter in the sed substitution in my opinion, because it also separates commands so you if you need to extend it in the future you might end up with `s;pattern;replacement;g;p;q...` which is not the most readable syntax.)

```suggestion
    | sed 's;@cilium/\([-_a-zA-Z0-9]*\)\(:\?\);`@cilium/\1 <https://github.com/orgs/cilium/teams/\1>`__\2;g' \
```

Reply:
> (As a side note, ; is not the best choice for a delimiter in the sed substitution in my opinion, because it also separates commands so you if you need to extend it in the future you might end up with s;pattern;replacement;g;p;q... which is not the most readable syntax.)

Any suggestions for a better delimiter? My usual delimiter is `/`, which is even worse, particularly for URLs. I think in the tree, `/` is the most common followed by `;`.

Reply:
Maybe `|` isn't too bad. From a quick look, it doesn't look like it's valid in a URL.

Reply:
If `/` is not suitable, I usually use `|` or `=` or possibly `@` depending on the context. I'm open to pretty much everything else, but I thought `;` works particularly badly so I'd raise it :)